lib/message_bus/http_client.rb

```diff
@@ -0,0 +1,324 @@
+require 'securerandom'
+require 'net/http'
+require 'json'
+require 'uri'
+require 'message_bus/http_client/channel'
+
+module MessageBus
+  # MessageBus client that enables subscription via long polling with support
+  # for chunked encoding. Falls back to normal polling if long polling is not
+  # available.
+  #
+  # @!attribute [r] channels
+  #   @return [Hash] a map of the channels that the client is subscribed to
+  # @!attribute [r] stats
+  #   @return [Stats] a Struct containing the statistics of failed and successful
+  #     polling requests
+  class HTTPClient
+    class InvalidChannel < StandardError; end
+    class MissingBlock < StandardError; end
+
+    attr_reader :channels,
+                :stats
+
+    attr_accessor :enable_long_polling,
```

Comment:
Can we document these or make them private?

Reply:
Oops don't have the habit of writing docs for yard so I missed this out. Fixed!